framework/tests/viewpointtest/ReceiverTypeNonNullTest.java

```diff
@@ -0,0 +1,17 @@
+// Test case for EISOP issue #782:
+// https://github.com/eisop/checker-framework/issues/782
+public class ReceiverTypeNonNullTest {
```

Comment:
How about two tests: `SuperConstructorCalls` and `ThisConstructorCalls` and have a few tests that ensure the corresponding constructor calls are correctly adapted?

Reply:
Added in the new commits.